projects/packages/my-jetpack/src/products/class-stats.php

```diff
@@ -162,23 +173,21 @@ public static function is_upgradable() {
 		if ( is_wp_error( $purchases_data ) ) {
 			return false;
 		}
+
 		if ( is_array( $purchases_data ) && ! empty( $purchases_data ) ) {
-			foreach ( $purchases_data as $purchase ) {
-				if (
-					(
-						// Purchase is Jetpack Stats...
-						str_starts_with( $purchase->product_slug, 'jetpack_stats' ) &&
-						// but not Jetpack Stats Free...
-						! str_contains( $purchase->product_slug, 'free' )
-					) || str_starts_with( $purchase->product_slug, 'jetpack_complete' )
-				) {
-					// Only Jetpack Stats paid plans should be eligible for this conditional.
-					// Sample product slugs: jetpack_stats_monthly
-					return false;
+			// For now, only the free and pwyw subs are upgradable
+			$upgradeable_stats_purchases = array_filter(
+				$purchases_data,
+				function ( $purchase ) {
+					return in_array( $purchase->product_slug, array( 'jetpack_stats_free', 'jetpack_stats_pwyw' ), true );
 				}
-			}
+			);
+
+			return ! empty( $upgradeable_stats_purchases );
 		}
-		return true;
+
+		// If there are no plans found, don't consider the product as upgradeable
+		return false;
```

Comment:
Just curious, what do you think the reasoning behind defaulting to `true` was before? I can't think of one, just want to make sure there isn't a good reason we are missing

Reply:
Good question -  I am not quite sure either 🤔 
It seems like the old approach was a blacklist approach and the new approach is more of a whitelist 🤷